app/controllers/application_controller.rb

```diff
@@ -1,5 +1,11 @@
 class ApplicationController < ActionController::Base
+  include Omniauth::Dsds::ControllerMethods
+
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  before_action :authenticate_user!
+
+  helper_method :current_user
```

Comment:
Could these not be part of the gem module? Use ActiveSupport::Concern and a included block